cmd/runner/main.go

```diff
@@ -1,56 +1,76 @@
 package main
 
 import (
-	"context"
+	"errors"
 	"flag"
 	"fmt"
+	"log"
 	"os"
 	"time"
 
+	"github.com/benchttp/runner/config"
+	"github.com/benchttp/runner/configparser"
 	"github.com/benchttp/runner/request"
 )
 
-const (
-	DefaultConcurrency = 1
-	DefaultRequests    = 0 // Use duration as exit condition if omitted.
-	DefaultDuration    = 60 * time.Second
-	DefaultTimeout     = 10 * time.Second
-)
-
 var (
-	url         string
-	concurrency int           // Number of connections to run concurrently.
-	requests    int           // Number of requests to run, use duration as exit condition if omitted.
-	duration    time.Duration // Duration of test, in seconds.
-	timeout     time.Duration // Timeout for each http request, in seconds.
+	configFile    string
+	url           string
+	concurrency   int           // Number of connections to run concurrently
+	requests      int           // Number of requests to run, use duration as exit condition if omitted.
+	timeout       time.Duration // Timeout for each http request
+	globalTimeout time.Duration // Duration of test
 )
 
-func parseArgs() {
-	url = os.Args[len(os.Args)-1]
+var defaultConfigFiles = []string{
+	"./.benchttp.yml",
+	"./.benchttp.yaml",
+	"./.benchttp.json",
+}
 
-	flag.IntVar(&concurrency, "c", DefaultConcurrency, "Number of connections to run concurrently")
-	flag.IntVar(&requests, "r", DefaultRequests, "Number of requests to run, use duration as exit condition if omitted")
-	flag.DurationVar(&duration, "d", DefaultDuration, "Duration of test, in seconds")
-	flag.DurationVar(&timeout, "t", DefaultTimeout, "Timeout for each http request, in seconds")
+func parseArgs() {
+	flag.StringVar(&configFile, "configFile", findFile(defaultConfigFiles), "Config file path")
+	flag.StringVar(&url, "url", "", "Target URL to request")
+	flag.IntVar(&concurrency, "concurrency", 0, "Number of connections to run concurrently")
+	flag.IntVar(&requests, "requests", 0, "Number of requests to run, use duration as exit condition if omitted")
+	flag.DurationVar(&timeout, "timeout", 0, "Timeout for each http request")
+	flag.DurationVar(&globalTimeout, "globalTimeout", 0, "Duration of test")
```

Comment:
Why do the flags needed to lose their shorthand ? it is said it was to match config file but to me it seems like the code could still really work with shorthand flags and be more dev-friendly

Also I still think expecting `url` as an argument and not a flag is better. The real reason is : it is the only on required as everything else can be defaulted in a sane and predictable manner (by the end user I mean, "yeah if I don't provide a number of workers it'll just be the default", not true for url).
Plus when we'll have others commands (ie `benchttp login`), it will make more sense for the command `run`/`test` to be passed an argument (`benchttp test http://myapp.foo/endpoint`).

Reply:
CLI options reflect the config file options, they need to share the same name for consistency. Consistency is more dev-friendly than shortcuts. See [TypeScript](https://www.typescriptlang.org/docs/handbook/compiler-options.html) or [Docker](https://docs.docker.com/engine/reference/commandline/cli/): no shortcuts needed.
We can still add shorthands later though, but this is the least of our priorities.

`url` can't be required in the CLI because it can be set in the config file, in which case we don't want to override it. This is how it was specced: we can still re-discuss it, but I suggest not here. Error handling for an empty url won't be an issue.

When we have other commands, we'll update the API as needed – now is not the time.